src/os/layer/image.js

```diff
@@ -422,6 +443,273 @@ os.layer.Image.prototype.getSVGSet = function() {
+ * grey.  A value of 1 will leave the contrast unchanged.  Other values are
+ * linear multipliers on the effect (and values over 1 are permitted).
+ *
+ * @override
+ * @param {number} value The contrast of the layer (values clamped between 0 and 2)
+ * @param {Object=} opt_options The layer options to use
+ */
+os.layer.Image.prototype.setContrast = function(value, opt_options) {
+  goog.asserts.assert(value >= 0 && value <= 2, 'contrast is not between 0 and 2');
+  var options = opt_options || this.layerOptions_;
+  if (options) {
+    options['contrast'] = value;
+    this.updateColorFilter();
+    os.style.notifyStyleChange(this);
+  }
+  os.layer.Image.base(this, 'setContrast', value);
+};
+
+
+/**
+ * Adjust layer saturation.  A value of 0 will render the layer completely
+ * unsaturated.  A value of 1 will leave the saturation unchanged.  Other
+ * values are linear multipliers of the effect (and values over 1 are
+ * permitted).
+ *
+ * @override
+ * @param {number} value The saturation of the layer (values clamped between 0 and 1)
+ * @param {Object=} opt_options The layer options to use
+ */
+os.layer.Image.prototype.setSaturation = function(value, opt_options) {
+  goog.asserts.assert(value >= 0, 'saturation is greater than 0');
+  var options = opt_options || this.layerOptions_;
+  if (options) {
+    options['saturation'] = value;
+    this.updateColorFilter();
+    os.style.notifyStyleChange(this);
+  }
+  os.layer.Image.base(this, 'setSaturation', value);
+};
+
+
+/**
+ * Adjust layer sharpness. A value of 0 will not adjust layer sharpness. A value of 1 will apply the maximum
+ * sharpness adjustment to the image.
+ *
+ * @override
+ * @param {number} value The sharpness of the layer (values clamped between 0 and 1)
+ * @param {Object=} opt_options The layer options to use
+ */
+os.layer.Image.prototype.setSharpness = function(value, opt_options) {
+  goog.asserts.assert(value >= 0 && value <= 1, 'sharpness is between 0 and 1');
+  var options = opt_options || this.layerOptions_;
+  if (options) {
+    options['sharpness'] = value;
+    this.updateColorFilter();
+    os.style.notifyStyleChange(this);
+  }
+  os.layer.Image.base(this, 'setSharpness', value);
+};
+
+
+/**
+ * Updates the color filter, either adding or removing depending on whether the layer is colored to a non-default
+ * color or colorized.
+ *
+ * @protected
+ */
+os.layer.Image.prototype.updateColorFilter = function() {
+  var source = this.getSource();
+  if (source instanceof ol.source.Image) {
+    if (this.getBrightness() != 0 || this.getContrast() != 1 || this.getSaturation() != 1 || this.getSharpness() != 0) {
+      // put the colorFilter in place if we are colorized or the current color is different from the default
+      source.addImageFilter(this.colorFilter_);
+    } else {
+      source.removeImageFilter(this.colorFilter_);
+    }
+  }
+};
+
+
+/**
+ * @return {?Array<!osx.ogc.ImageStyle>}
+ */
+os.layer.Image.prototype.getStyles = function() {
```

Comment:
Styles are specific to tile layers, where servers like WMS provide preset styles that may be applied to the layer. These styles change appearance of the tiles server-side. Image layers do not have a similar specification, so we do not need to duplicate all of that code here. We're only interested in the client-side transforms for brightness, contrast, etc.